### ISSUE TITLE
Clean up terminology and better explain tradeoffs

### ISSUE THREAD

**Issue by cpaasch** [network-quality/draft-cpaasch-ippm-responsiveness#34]
We are using a confusing mix of "working conditions", "saturation", "fully loaded", "busy as possible", "typical day-to-day pattern",... This needs to be cleaned up!

We should start probably with a description of what "working conditions" means. Why it is a near worst-case while still being realistic. What the tradeoffs here are. And, that it is going to evolve over time as typical day-to-day patterns are evolving.

Feedback from Al Morton (@acmacm):

> On Dec 17, 2021, at 3:50 PM, MORTON JR., AL <acmorton@att.com> wrote:
> 
> Hi authors and ippm-chairs,
>  
> Thanks for writing this-up!
>  
> I took one pass through, and have the following comments during Adoption call for draft-cpaasch-ippm-responsiveness:
>  
> TL;DR:
> Many previously undefined terms were used here, and a more direct description using the term “saturation” seems possible, IMO.
> 
> I fully agree with you. We are not very good at describing the "working conditions"/"saturation" we are aiming for, why we use these and what the right approach is.
> 
> The type of "working conditions" is crucial as to what the measurement result will be
> For example, flooding the network with UDP traffic will saturate the network pretty well, but it is far away from a realistic working condition.
> 
> What we are aiming for is near worst-case scenario while still being realistic. At least, that is the intention and it may be good to have an open discussion about this.
> 
> IPPM has used a template for metric drafts, and use of the hierarchy of singleton, sample, and statistic metrics from RFC 2330 will help with clarity/answer many of my questions.
>  
> regards (I’m off-line for a while now, so enjoy the holidays),
> Al
>  
> From the Abstract:
>  
>    This document specifies the "RPM Test" for measuring responsiveness.
>    It uses common protocols and mechanisms to measure user experience
>    especially when the network is fully loaded ("responsiveness under
>    working conditions".)  The measurement is expressed as "Round-trips
>    Per Minute" (RPM) and should be included with throughput (up and
>    down) and idle latency as critical indicators of network quality.
>  
> “fully loaded” and “working conditions” aren’t necessarily the same, to me.  I’ll be looking for better definitions.
> 
> Agree'd.
> 
>  
> 3.  Goals
>  
>    The algorithm described here defines an RPM Test that serves as a
>    good proxy for user experience.  This means:
>  
>    1.  Today's Internet traffic primarily uses HTTP/2 over TLS.  Thus,
>        the algorithm should use that protocol.
>  
>        As a side note: other types of traffic are gaining in popularity
>        (HTTP/3) and/or <???? UDP ???> are already being used widely (RTP).
>  
> There are many measurement stability challenges when TCP is involved, see section 4 of RFC8337:  https://datatracker.ietf.org/doc/html/rfc8337#section-4
> RFC8337 intentionally broke the TCP control loop to make measurements in the face of these challenges.
> 
> Yes, we are aware of these kind of stability challenges. And actually sometimes observe them as results can vary to some degree across different runs.
> 
> The goal is to get as close as possible to a stable measurement result, while still using the protocols the end-users use on a day-to-day basis.
> 
>  
> 4.1.  Working Conditions
>  
>    For the purpose of this methodology, typical "working conditions"
>    represent a state of the network in which the bottleneck node is
>    experiencing ingress and egress flows similar to those created by
>    humans in the typical day-to-day pattern.
>  
>    While a single HTTP transaction might briefly put a network into
>    working conditions, making reliable measurements requires maintaining
>    the state over sufficient time.
>  
>    The algorithm must also detect when the network is in a persistent
>    working condition, also called "saturation".
>  
>    Desired properties of "working condition":
>  
>    o  Should not waste traffic, since the person may be paying for it
>  
>    o  Should finish within a short time to avoid impacting other people
>       on the same network, to avoid varying network conditions, and not
>       try the person's patience.
>  
> These seem like reasonable goals for the traffic that loads the network.
> New terms needing definition were introduced:
> “persistent working condition = saturation”,
> which is different from
> “ingress and egress flows similar to those created by humans in the typical day-to-day pattern”
>  
> Later in 4.1.1, terms like “saturate a path”  and “fill the pipe” appear, and
>  
>    The goal of the RPM Test is to keep the network as busy as possible
>    in a sustained and persistent way.  It uses multiple TCP connections
>    and gradually adds more TCP flows until saturation is reached.
>  
> The terms “busy as possible”, and “typical day-to-day pattern”, or
> “saturation” and “working conditions” indicate different load levels to me.
>  
> @@@@ Suggestion: I think it would help to simplify the terminology in this draft. You intend to measure a saturated path, so just say that. No “typical”, no “working conditions”, etc., in these early sections. 
>  
> The sentence beginning “The goal...” should really appear in Section 3. Goals
>  
> Also, you have defined a measurement method in the sentence, “It uses...” above. This method of adding connections has been observed in other measurement systems, but it isn’t typical of user traffic, especially when each connection has an ~infinite amount of data to send during the test.
> 
> From your comments I see that we definitely need a longer explanation of the tradeoffs that are being made of measuring this near worst-case, but realistic scenario. As you correctly point out, we are mixing confusing and sometimes contradictory terms. This needs to be cleaned up.
